include/deal.II/base/aligned_vector.h

```diff
@@ -1553,6 +1553,10 @@ AlignedVector<T>::replicate_across_communicator(const MPI_Comm &   communicator,
   used_elements_end      = elements.get() + array_size;
   allocated_elements_end = used_elements_end;
 
+  // Make sure the pointers are correctly set for all processes before anyone
+  // accesses the shared data
+  MPI_Barrier(communicator);
```

Comment:
I think you should be able to get away with doing this on the `shmem_group_communicator` instead of the global one.